sw/device/lib/dif/dif_uart.c

```diff
@@ -77,6 +79,19 @@ static ptrdiff_t uart_irq_offset_get(dif_uart_interrupt_t irq_type) {
   return offset;
 }
 
+static void uart_reset(const dif_uart_t *uart) {
+  mmio_region_write32(uart->base_addr, UART_CTRL_REG_OFFSET, 0u);
+  // Write to the relevant bits clears the FIFOs
+  mmio_region_write32(
+      uart->base_addr, UART_FIFO_CTRL_REG_OFFSET,
+      (1u << UART_FIFO_CTRL_RXRST) | (1u << UART_FIFO_CTRL_TXRST));
+  mmio_region_write32(uart->base_addr, UART_OVRD_REG_OFFSET, 0u);
+  mmio_region_write32(uart->base_addr, UART_TIMEOUT_CTRL_REG_OFFSET, 0u);
+  mmio_region_write32(uart->base_addr, UART_INTR_ENABLE_REG_OFFSET, 0u);
+  mmio_region_write32(uart->base_addr, UART_INTR_STATE_REG_OFFSET,
+                      UART_INTR_STATE_MASK);
```

Comment:
How about writing `0xffffu` instead of `0xffu`? This way you wouldn't have to hardcode the flags and the code would work as intended even if new flags are added in the future.

Reply:
Sounds good, might as well then make it 0xffffffff.

Reply:
Oops, yes that's what I meant, thanks!